src/blocks/weather.rs

```diff
@@ -47,23 +51,34 @@
-//! `location`        | Location name (exact format depends on the service)                | Text   | -
-//! `temp`            | Temperature                                                        | Number | degrees
-//! `apparent`        | Australian Apparent Temperature                                    | Number | degrees
-//! `humidity`        | Humidity                                                           | Number | %
-//! `weather`         | Textual brief description of the weather, e.g. "Raining"           | Text   | -
-//! `weather_verbose` | Textual verbose description of the weather, e.g. "overcast clouds" | Text   | -
-//! `wind`            | Wind speed                                                         | Number | -
-//! `wind_kmh`        | Wind speed. The wind speed in km/h                                 | Number | -
-//! `direction`       | Wind direction, e.g. "NE"                                          | Text   | -
+//!  Key                | Value                                                              | Type   | Unit
+//! --------------------|--------------------------------------------------------------------|--------|-----
+//! `icon`              | Icon representing the weather                                      | Icon   | -
+//! `location`          | Location name (exact format depends on the service)                | Text   | -
+//! `temp`              | Temperature                                                        | Number | degrees
+//! `apparent`          | Australian Apparent Temperature                                    | Number | degrees
+//! `humidity`          | Humidity                                                           | Number | %
+//! `weather`           | Textual brief description of the weather, e.g. "Raining"           | Text   | -
+//! `weather_verbose`   | Textual verbose description of the weather, e.g. "overcast clouds" | Text   | -
+//! `wind`              | Wind speed                                                         | Number | -
+//! `wind_kmh`          | Wind speed. The wind speed in km/h                                 | Number | -
+//! `direction`         | Wind direction, e.g. "NE"                                          | Text   | -
+//! `temp_forcast`      | Temperature forcast                                                | Number | degrees
+//! `apparent_forcast`  | Australian Apparent Temperature forcast                            | Number | degrees
+//! `humidity_forcast`  | Humidity forcast                                                   | Number | %
+//! `wind_forcast`      | Wind speed forcast                                                 | Number | -
+//! `wind_kmh_forcast`  | Wind speed forcast. The wind speed in km/h                         | Number | -
+//! `direction_forcast` | Wind direction forcast, e.g. "NE"                                  | Text   | -
```

Comment:
Notable omission is weather_forcast / weather_verbose_forcast. I suppose the most common in the sampled forecasts could be used...

Reply:
Also `icon_forecast`. I'm not familiar with openweathermap's API (and I couldn't even found the docs for `https://api.openweathermap.org/data/2.5/forecast` on their site), but at least, from my understanding, it is easy to implement it in metno? I'm not against having some placeholders available for some providers but not other.

Reply:
The api doc is here: https://openweathermap.org/forecast5

both owm and metno have descriptions of the weather for each interval, but not a summary. That's why I had omitted the `weather_forecast`, `weather_verbose_forecast`, and `icon_forecast`.

Do you think taking the most common descriptions over the sampled intervals is good enough or should it be left out as it is perhaps misleading? For example another strategy would be to show the worst/most severe weather but that is a bit nebulous :drum:

Reply:
Ha! I just realized that my understanding of this feature is completely wrong. I thought that `forecast_hours = 3` will show me the weather _after_ three hours. What it actually does is it shows the "average" weather throughout the next three hours :facepalm:

Reply:
Do you have a better key name or description we can use to clarify?

Reply:
Maybe we can have something like `x_forecast_{avg,min,max,final}`? This allows for `{icon,weather,weather_verbose}_forecast_final`. That's a lot of placeholders, but we don't have to list all of them in the docs, just explain how they are formed with a few examples.

Reply:
What's the average of tornado and clear skies? Strong winds?

Reply:
IDK. I don't think providing min/max/avg for weather/icon is a good idea.

Reply:
I agree

Reply:
Any thoughts no this? I can imagine

```toml
format_alt = " $icon_ff in 12 hours: $temp_fa($temp_fmin-$temp_fmax) "
```

being useful. Here `_ff` stands for "forecast final", `_fa` for "forecast average" and `_f{min,max}` for "forecast {min,max}". That's just one way to name them.

Reply:
I think this is a good idea. 
`icon`, `weather`, and `weather_verbose` can only be suffixed with `_ffin`.
`temp`, `apparent`, `humidity`, `wind`, `wind_kmh`, `direction` can be suffixed with `_f{fin,avg,min,max}`